components/api/src/main/java/com/hotels/styx/api/StyxCoreObservable.java

```diff
@@ -62,31 +50,23 @@ public StyxCoreObservable(CompletionStage<T> future) {
         return new StyxCoreObservable<T>(Observable.error(cause));
     }
 
-    public <U> StyxObservable<U> map(Function<T, U> transformation) {
+    public <U> StyxObservable<U> map(Function<? super T, ? extends U> transformation) {
```

Comment:
I'd rather change the return type name to R (instead of U) following Java 8 (e.g,  https://docs.oracle.com/javase/8/docs/api/java/util/function/Function.html) and Observable's convention.  
Maybe just copy the generic names from the Observable interface for the same methods?